### PR TITLE
[MIN-56] Use Seldon Core directly for model deployment

### PR DESCRIPTION
**Pull request by d-lowl**
* Add Seldon Deployment manifest for an LLM (defaults to summarization pipeline and flan-t5 model)
* Add documentation on prerequisites (K8s, Seldon Core), deployment, querying and using in the app
* Update the app to use this model
* Remove zenml depoyment pipeline

TODO:

- [x] Infer query endpoint
- [x] Update main READMEs
- [x] Update the frontend up to call the correct endpoint
- [x] Remove deployment pipeline

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -37,23 +37,15 @@ def _get_prediction_endpoint() -> Optional[str]:
...
+    if ingress_ip is None:
         return None
+    return f"http://{ingress_ip}/seldon/matcha-seldon-workloads/llm/v2/models/transformer/infer"
 
 
 def _create_payload(
     messages: List[Dict[str, str]]
-) -> Dict[str, Dict[str, List[Dict[str, str]]]]:
+) -> Dict[str, List[Dict[str, str]]]:
     """Create a payload from the user input to send to the LLM model.
```
Comment (Christopher-Norman) [inline app/app.py:49]: Docstring update to reflect new output type of `Dict[str, List[Dict[str, str]]]`

File: app/app.py
Comment (Christopher-Norman) [inline app/app.py:1]: When it comes to integrating the Vector Database, would we require a custom Seldon image OR would we have to perform the context document retrieval within the Streamlit app?

**Review by Christopher-Norman — APPROVED** [review]
LGTM, one docstring update and one general comment about integrating the vector DB

File: app/app.py
Comment (d-lowl) [reply app/app.py:1]: I's say we keep the model deployment as is, and call the database from the streamlit app. (Later we might want to have a logic service that would call both the vector store and the model, and will be called by the frontend; but there's no need right now)

File: app/app.py (hunk 1 above)
Comment (d-lowl) [reply app/app.py:49]: :+1:

File: app/app.py
Comment (Christopher-Norman) [reply app/app.py:1]: Sounds good to me. Tagging @dudeperf3ct for visibility and a question, we should add an ingress to the Chroma DB server, right?

Comment (archena) [reply app/app.py:1]: I thought we'd have an intermediary service as @d-lowl suggests. But that also feels like unnecessary complexity at this stage. And in that case, indeed we would want an ingress for Chroma.

Edit: everything deployed to k8s, no need for ingress other than for Streamlit!

Comment (dudeperf3ct) [reply app/app.py:1]: There's a bit of change to the type of task for the seldon model as well. We are not using a summarization model but a text generation model or any of the seldon supported types that is same as LLM based model. The implementation in the frontend would be, we will fetch revelant documents from the chroma store, construct a custom prompt using documents from chroma as context and pass this to the deployed model. The deployed model should take in this prompt as input and output a answer text.

For simplicity adding logic to streamlit itself sounds good.

Comment (d-lowl) [reply app/app.py:1]: @dudeperf3ct 
There are two ways here:
* Just use text-generation task, and construct a correct flan-t5 instruction prompt ourselves
* Or use QA task, if it's supported by flan-t5 

We've just discussed it with Matt, I'll try out the second option now, to see if it's viable

File: README.md [hunk 2]
```diff
@@ -42,25 +42,53 @@ pip install matcha-ml
 matcha provision
 ```
 
-After the provisioning completes, run the following bash script
+After the provisioning completes, we will have on hand these resources:
+* Kubernetes cluster on Azure
+* Seldon Core installed on this cluster
+* Istio ingress installed on this cluster
```
Comment (dudeperf3ct) [inline README.md:48]: Question: Do we need a ingress for seldon? Can streamlit not access seldon service in same way to chroma server and we just add one ingress on top of streamlit?

File: README.md [hunk 3]
```diff
@@ -42,25 +42,53 @@ pip install matcha-ml
...
+
+The expected payload structure is as follows:
+```json
+{
+    "inputs": [
+        {
+            "name": "array_inputs",
+            "shape": [-1],
+            "datatype": "string",
+            "data": "String to summarize"
```
Comment (dudeperf3ct) [inline README.md]: I think `data` payload should be changed as it's a text-to-text generation task.

**Review by dudeperf3ct — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (d-lowl) [reply README.md:48]: Yeah, we can talk to Seldon on k8s directly. I wouldn't disable ingress until streamlit app is deployed to the cluster

File: README.md (hunk 3 above)
Comment (dudeperf3ct) [reply README.md]: It's also worth mentioning the format expected by flan-t5 model to be a question-answering task.

Comment (d-lowl) [reply README.md]: :+1:

**Review by JonoCX — APPROVED** [review]
LGTM - nice work! 🦖

**Pull request merged**